src/main/java/org/apache/mesos/specification/TaskTypeSpecification.java

```diff
@@ -1,8 +1,16 @@
 package org.apache.mesos.specification;
 
+import org.apache.mesos.Protos;
+
+import java.util.Collection;
+
 /**
- * A TaskTypeSpecification describes the count of a particular TaskSpecification which should be launched.
+ * A TaskTypeSpecification describes a Type of Task and how many should be launched.
  */
-public interface TaskTypeSpecification extends TaskSpecification {
+public interface TaskTypeSpecification {
```

Comment:
Oh, I didn't address the final point.  The developer definitely owns the creation of `TaskTypeSpecification` objects.  Without ID input, it's not possible, as far as I can see to turn `TaskTypeSpecification`s into `TaskSpecification`s.  Open to ideas.